perun-core/src/main/java/cz/metacentrum/perun/core/api/UsersManager.java

```diff
@@ -895,92 +895,121 @@ void changePassword(PerunSession sess, User user, String loginNamespace, String
 	void changeNonAuthzPassword(PerunSession sess, String token, String password, String lang) throws UserNotExistsException, LoginNotExistsException, PasswordChangeFailedException, PasswordOperationTimeoutException, PasswordStrengthFailedException, InvalidLoginException, PasswordStrengthException, PasswordResetLinkExpiredException, PasswordResetLinkNotValidException;
 
 	/**
-	 * Reserves random password in external system. User must not exists.
+	 * Reserves random password in external system. User must not exist.
 	 *
-	 * @param sess
-	 * @param user
-	 * @param loginNamespace
+	 * @param sess Perun session
+	 * @param user User to reserve password for
+	 * @param loginNamespace Login-namespace to reserve password in
 	 * @throws InternalErrorException
 	 * @throws PasswordCreationFailedException
-	 * @throws UserNotExistsException
-	 * @throws LoginNotExistsException
+	 * @throws UserNotExistsException When the user doesn't exist
+	 * @throws LoginNotExistsException When user doesn't have login in specified namespace
 	 */
 	void reserveRandomPassword(PerunSession sess, User user, String loginNamespace) throws PasswordCreationFailedException, PrivilegeException, UserNotExistsException, LoginNotExistsException, PasswordOperationTimeoutException, PasswordStrengthFailedException, InvalidLoginException;
 
 	/**
-	 * Reserves the password in external system. User must not exists.
+	 * Reserves the password in external system. User must not exist.
 	 *
-	 * @param sess
-	 * @param userLogin string representation of the userLogin
-	 * @param loginNamespace
-	 * @param password
+	 * @param sess Perun session
+	 * @param userLogin String representation of the userLogin
+	 * @param loginNamespace Login-namespace to reserve password in
+	 * @param password Password to be reserved
 	 * @throws InternalErrorException
 	 * @throws PasswordCreationFailedException
-	 * @throws InvalidLoginException
+	 * @throws InvalidLoginException When login of user has invalid syntax (is not allowed)
 	 */
 	void reservePassword(PerunSession sess, String userLogin, String loginNamespace, String password)
 			throws PasswordCreationFailedException, PrivilegeException, PasswordOperationTimeoutException, PasswordStrengthFailedException, InvalidLoginException, PasswordStrengthException;
 
 	/**
-	 * Reserves the password in external system. User must exists.
+	 * Reserves the password in external system. User must exist.
```

Comment:
Please add also that login for that user in the specified namespace must exist in Perun when calling this method.

Reply:
Added for all